### ISSUE TITLE
scipy.special.tests.test_orthogonal::test_roots_jacobi fails on 32-bit intel

### ISSUE THREAD

**Issue by mwhudson** [scipy/scipy#9247]
### Reproducing code example:
```
python3.6 runtests.py -v -t scipy.special.tests.test_orthogonal::test_roots_jacobi
```

In one of those "oh dear" sort of ways, it does *not* fail in a debug build. So it's probably some rounding thing caused by the C compilers optimization settings.

### Error message:
https://gist.github.com/mwhudson/f40be8bc89eea32a469257d3d9cb478c

### Scipy/Numpy/Python version information:
```
1.2.0.dev0+641bc92 1.14.5 sys.version_info(major=3, minor=6, micro=6, releaselevel='final', serial=0)
```

**Comment by mwhudson** [comment]
Ubuntu compiles 32-bit things to use x87 arithmetic, does your system use sse2 by any chance? There are also many other compiler flags that can make a difference... if just increasing the tolerance is OK I can certainly have a go at that.

**Comment by rgommers** [comment]
Yes, IIRC my setup did use SSE2. Yes, if the test passes with reasonable increase in test tolerance, that should be OK.
